hs_core/hydroshare/users.py

```diff
@@ -611,13 +624,22 @@ def get_resource_list(creator=None,
                     try:
                         owner = user_from_id(owner, raise404=False)
                     except User.DoesNotExist:
-                        queries[t].append(Q(owners__isnull=True))
+                        pass
                     else:
-                        queries[t].append(Q(owners=owner))
+                        queries[t].append(Q(raccess__r2urp__privilege__lte=PrivilegeCodes.OWNER,
+                                            raccess__r2urp__user=owner.uaccess))
+
                         if user != owner:
                             public = True
                 else:
-                    queries[t].append(Q(edit_users=user) | Q(view_users=user) | Q(owners=user) | Q(public=True))
+                    queries[t].append(Q(raccess__r2urp__privilege__lte=PrivilegeCodes.CHANGE,
+                                        raccess__r2urp__user=user.uaccess) |
+                                      Q(raccess__r2urp__privilege__lte=PrivilegeCodes.VIEW,
+                                        raccess__r2urp__user=user.uaccess) |
+                                      Q(raccess__r2urp__privilege__lte=PrivilegeCodes.OWNER,
+                                        raccess__r2urp__user=user.uaccess) |
+                                      Q(raccess__public=True)
+                                      )
```

Comment:
Yes, in this case something like: `queries[t].append(Q(raccess__resource__in=u.uaccess.get_held_resources()))`